### PR TITLE
Adjust pom plugin position and add docs

### PR DESCRIPTION
**Pull request by kezhenxu94**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance
- [x] polish/documentation

- Related issues

#2864

### CODE REVIEW

File: docs/en/guides/How-to-build.md [hunk 1]
```diff
@@ -142,5 +142,24 @@ a oap image with name `bar/oap:foo`, run the following command
 ```
 5. Run `./mvnw compile -Dmaven.test.skip=true`
 6. Run `maven update`. Must remove the clean projects item before maven update(This will be clear the proto conversion Java file generated by the complie)
-7. Run `./mvnw compile` complie collector-remote-grpc-provider and apm-protocol
+7. Run `./mvnw compile` compile collector-remote-grpc-provider and apm-protocol
 8. Refresh project
+
+## Write necessary tests and run them
```
Comment (wu-sheng) [inline docs/en/guides/How-to-build.md]: Add this to here https://github.com/apache/skywalking/blob/master/docs/en/guides/README.md#for-code-developer . Make sense?

Comment (kezhenxu94) [reply docs/en/guides/How-to-build.md]: Added

Comment (wu-sheng) [reply docs/en/guides/How-to-build.md]: I think we don't need this paragraph here. Build source doesn't mean write codes and contribute to upstream.

File: docs/en/guides/How-to-build.md [hunk 2]
```diff
@@ -142,5 +142,24 @@ a oap image with name `bar/oap:foo`, run the following command
...
+
+After setting up the environment and writing your codes, to make it more easily accepted by SkyWalking project, you'll
```
Comment (wu-sheng) [inline docs/en/guides/How-to-build.md]: in order to make it

File: docs/en/guides/How-to-build.md (hunk 1 above)
Comment (wu-sheng) [reply docs/en/guides/How-to-build.md]: I suggest move all to there

Comment (kezhenxu94) [reply docs/en/guides/How-to-build.md]: Got it, I misunderstood your meaning, I thought you're asking me to put the link there, what a embarrass :-)

**Pull request merged**